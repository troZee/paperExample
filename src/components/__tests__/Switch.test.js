import * as React from 'react';
import renderer from 'react-test-renderer';
import Switch from '../Switch.tsx';
import { pink500 } from '../../styles/colors.tsx';

it('renders on switch', () => {
  const tree = renderer.create(<Switch value />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders off switch', () => {
  const tree = renderer.create(<Switch value={false} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders disabled switch', () => {
  const tree = renderer.create(<Switch disabled value />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders switch with color', () => {
  const tree = renderer.create(<Switch value color={pink500} />).toJSON();

  expect(tree).toMatchSnapshot();
});
