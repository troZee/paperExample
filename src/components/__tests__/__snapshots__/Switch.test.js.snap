// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders disabled switch 1`] = `
<RCTSwitch
  accessibilityRole="button"
  disabled={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#03dac4"
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  tintColor=""
  value={true}
/>
`;

exports[`renders off switch 1`] = `
<RCTSwitch
  accessibilityRole="button"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#03dac4"
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  tintColor=""
  value={false}
/>
`;

exports[`renders on switch 1`] = `
<RCTSwitch
  accessibilityRole="button"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#03dac4"
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  tintColor=""
  value={true}
/>
`;

exports[`renders switch with color 1`] = `
<RCTSwitch
  accessibilityRole="button"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#e91e63"
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  tintColor=""
  value={true}
/>
`;
